cmd/snap/cmd_snap_op.go

```diff
@@ -353,6 +381,60 @@ func (x *cmdInstall) Execute([]string) error {
 	return showDone([]string{name}, "install")
 }
 
+func (x *cmdInstall) installMany(names []string) error {
+	// sanity check
+	for _, name := range names {
+		if strings.Contains(name, "/") || strings.HasSuffix(name, ".snap") || strings.Contains(name, ".snap.") {
+			return fmt.Errorf("only a single snap file can be installed")
+		}
+	}
+
+	cli := Client()
+	changeID, err := cli.InstallMany(names)
+	if err != nil {
+		return err
+	}
+
+	chg, err := wait(cli, changeID)
+	if err != nil {
+		return err
+	}
+
+	var installed []string
+	if err := chg.Get("snap-names", &installed); err != nil && err != client.ErrNoData {
+		return err
+	}
+
+	if len(installed) > 0 {
+		return showDone(installed, "install")
+	}
+
+	return nil
+}
+
+func (x *cmdInstall) Execute([]string) error {
+	if err := x.setChannelFromCommandline(); err != nil {
+		return err
+	}
+	if err := x.validateMode(); err != nil {
+		return err
+	}
+
+	dangerous := x.Dangerous || x.ForceDangerous
+	if len(x.Positional.Snaps) == 1 {
+		opts := &client.SnapOptions{
+			Channel:   x.Channel,
+			DevMode:   x.DevMode,
+			JailMode:  x.JailMode,
+			Revision:  x.Revision,
+			Dangerous: dangerous,
```

Comment:
Right, this is currently not done. The current refreshMany (and installMany, removeMany) will refuse to take any options. I think this is a bug at least for install because it seems ok to do "snap install --beta one two" and get both from beta. However for "snap remove --revision=17 one two" does not make much sense. So this needs some more work (either here or in  a follwoup branch, your call).